src/common/device_helpers.cuh

```diff
@@ -1179,4 +1184,70 @@ auto Reduce(Policy policy, InputIt first, InputIt second, Init init, Func reduce
   }
   return aggregate;
 }
+
+template <bool accending, typename IdxT, typename U>
+void ArgSort(xgboost::common::Span<U> values, xgboost::common::Span<IdxT> sorted_idx) {
+  size_t bytes = 0;
```

Comment:
Maybe get into the habit of checking if cub input length is < 2^31. Let's try to avoid our previous troubles with cub.

Reply:
Done.